tests/rkt_tests.go

```diff
@@ -515,3 +557,204 @@ func checkAppStatus(t *testing.T, ctx *rktRunCtx, multiApps bool, appName, expec
+				if len(fields) != 2 {
+					t.Fatalf("Unexpected network info format: %v", n)
+				}
+
+				ip4 := strings.Split(fields[1], "=")
+				if len(ip4) != 2 {
+					t.Fatalf("Unexpected network info format: %v", n)
+				}
+
+				networkName := fields[0]
+				p.networks[networkName] = &networkInfo{
+					name: networkName,
+					ipv4: ip4[1],
+				}
+			}
+		case "pid":
+			pid, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				t.Fatalf("Cannot parse the pod's pid %q: %v", tuples[1], err)
+			}
+			p.pid = pid
+		}
+		if strings.HasPrefix(tuples[0], "app-") {
+			exitCode, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				t.Fatalf("cannot parse exit code from %q : %v", tuples[1], err)
+			}
+			appName := strings.TrimPrefix(tuples[0], "app-")
+
+			for _, app := range p.apps {
+				if app.name == appName {
+					app.exitCode = exitCode
+					break
+				}
+			}
+		}
+	}
+}
+
+func getPodDir(t *testing.T, ctx *rktRunCtx, podID string) string {
+	podsDir := path.Join(ctx.dataDir(), "pods")
+
+	dirs, err := ioutil.ReadDir(podsDir)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	for _, dir := range dirs {
+		podDir := path.Join(podsDir, dir.Name(), podID)
+		if _, err := os.Stat(podDir); err == nil {
+			return podDir
+		}
+	}
+	t.Fatalf("Failed to find pod directory for pod %q", podID)
+	return ""
+}
+
+// getPodInfo returns the pod info for the given pod ID.
+func getPodInfo(t *testing.T, ctx *rktRunCtx, podID string) *podInfo {
+	p := &podInfo{
+		id:       podID,
+		apps:     make(map[string]*appInfo),
+		networks: make(map[string]*networkInfo),
+	}
+
+	// Read pod manifest.
+	podManifestPath := path.Join(getPodDir(t, ctx, podID), "pod")
+	f, err := os.Open(podManifestPath)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	p.manifest, err = ioutil.ReadAll(f)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Fill app infos.
+	var manifest schema.PodManifest
+	if err := json.Unmarshal(p.manifest, &manifest); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	for _, app := range manifest.Apps {
+		appName := app.Name.String()
+		p.apps[appName] = &appInfo{
+			name: appName,
+			// TODO(yifan): Get the image's name.
+			image: &imageInfo{id: app.Image.ID.String()},
+		}
+	}
+
+	// Fill other infos.
+	output, err := exec.Command("/bin/bash", "-c", fmt.Sprintf("%s status %s", ctx.cmd(), podID)).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	parsePodInfoOutput(t, string(output), p)
+
+	return p
+}
+
+// parseImageInfoOutput parses the 'rkt image list' result into imageInfo struct.
+// For example, the 'result' can be:
+// 'sha512-e9b77714dbbfda12cb9e136318b103a6f0ce082004d09d0224a620d2bbf38133 nginx:latest 2015-10-16 17:42:57.741 -0700 PDT true'
+func parseImageInfoOutput(t *testing.T, result string) *imageInfo {
+	fields := strings.Split(result, "\t")
+	if len(fields) != 4 {
+		t.Fatalf("Failed to parse result: %q", result)
```

Comment:
@alban Never mind, strings.Fields() can't work because there are spaces in the import time.